### ISSUE TITLE
2.0.1 dataclasses have a incorrect types that don't match carnality 

### ISSUE THREAD

**Issue by T-amairi** [mobilityhouse/ocpp#340]
Based on the OCPP [spec](https://raw.githubusercontent.com/mobilityhouse/ocpp/master/docs/v201/OCPP-2.0.1_part2_specification.pdf) (page 379),  IdTokenType should have an **optional list** for **additional_info** attribute (see the cardinality of **additionalInfo** in the spec : 0..*). 

- Current version (0..1) : 
````python
@dataclass
class IdTokenType:
    """
    Contains a case insensitive identifier to use for the authorization and the
    type of authorization to support multiple forms of identifiers.
    IdTokenType is used by: AuthorizationData, Common:IdTokenInfoType,
    RequestStartTransactionRequest, AuthorizeRequest, TransactionEventRequest,
    ReserveNowRequest, CustomerInformationRequest
    """

    id_token: str
    type: enums.IdTokenType
    additional_info: Optional[AdditionalInfoType] = None
````

- Correct version (0..*) ? :
````python
@dataclass
class IdTokenType:
    """
    Contains a case insensitive identifier to use for the authorization and the
    type of authorization to support multiple forms of identifiers.
    IdTokenType is used by: AuthorizationData, Common:IdTokenInfoType,
    RequestStartTransactionRequest, AuthorizeRequest, TransactionEventRequest,
    ReserveNowRequest, CustomerInformationRequest
    """

    id_token: str
    type: enums.IdTokenType
    additional_info: Optional[List[AdditionalInfoType]] = None
````

**Comment by Jared-Newell-Mobility** [comment]
Thank-you for raising this; so from the schema snippet below IdTokenType is defined as being an array, however, there appears to be inconsistency with type declarations - I'll change this to a bug. 

`	"IdTokenType": {
	  "description": "Contains a case insensitive identifier to use for the authorization and the type of authorization to support multiple forms of identifiers.\r\n",
	  "javaType": "IdToken",
	  "type": "object",
	  "additionalProperties": false,
	  "properties": {
		"customData": {
		  "$ref": "#/definitions/CustomDataType"
		},
		"additionalInfo": {
		  "type": "array",
		  "additionalItems": false,
		  "items": {
			"$ref": "#/definitions/AdditionalInfoType"
		  },
		  "minItems": 1
		},
		"idToken": {
		  "description": "IdToken is case insensitive. Might hold the hidden id of an RFID tag, but can for example also contain a UUID.\r\n",
		  "type": "string",
		  "maxLength": 36
		},
		"type": {
		  "$ref": "#/definitions/IdTokenEnumType"
		}
	  },
	  "required": [
		"idToken",
		"type"
	  ]
	},`

**Comment by Jared-Newell-Mobility** [comment]
PR approved in https://github.com/mobilityhouse/ocpp/pull/529 will be included in v1.0.0
